### PR TITLE
feat(providers/microsoft): add DefaultAzureCredential support to AzureContainerVolumeHook

### PR DESCRIPTION
**Pull request by Lee-W**
---
**^ Add meaningful description above**
Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/providers/microsoft/azure/hooks/container_volume.py [hunk 1]
```diff
@@ -54,14 +56,22 @@ def _get_field(self, extras, name):
...
                 lazy_gettext("Blob Storage Connection String (optional)"), widget=BS3PasswordFieldWidget()
             ),
+            "subscription_id": StringField(
+                lazy_gettext("Subscription ID (optional)"),
+                widget=BS3TextFieldWidget(),
+            ),
+            "resource_group": StringField(
+                lazy_gettext("Resource group name (optional)"),
+                widget=BS3TextFieldWidget(),
+            ),
```
Comment (Taragolis) [inline airflow/providers/microsoft/azure/hooks/container_volume.py:74]: Would be nice to have information about this extra fields into the documentation 

https://github.com/apache/airflow/blob/b82ce61285f3f4f0c7eccb2c3effaef53c9fb84e/docs/apache-airflow-providers-microsoft-azure/connections/azure_container_volume.rst?plain=1#L60-L64

Comment (Lee-W) [reply airflow/providers/microsoft/azure/hooks/container_volume.py:74]: Sure! Just updated these 2 fields

**Review by pankajastro — APPROVED** [review]

**Comment by Lee-W** [comment]
There's a mypy error on `airflow/providers/amazon/aws/hooks/appflow.py:27`, which is not changed in this PR. Is there I can do for fixing this?

**Comment by potiuk** [comment]
This is because your PR generally changes dependencies and it means that it triggers eager upgrade - which means that we had to resolve and find new set of dependencies that your new dependencies and Airflow in general sees as consistent set of dependencies.

Generally follow the advice and attempt to fix it:

image

The thing is that another dependency that is being upgraded adds extra information or changes the definition of the other thing that causes mypy to see it as an issue.

The best way to fix it is to figure out how to fix it and submit a separate PR to fix it and then rebase your PR. Or you can wait until someone else fixes it. This what happens now with main builds (canary builds) - see here:

https://github.com/apache/airflow/actions/runs/6011017073/job/16304198261

Whoever fixes it - fixes it for the other as well. The fastest way to deal with it is to fix it yourself.

**Comment by Lee-W** [comment]
Thanks! I couldn't reproduce it locally yesterday and thus feel a bit confused. I'll take a deeper look next time.

**Review by pankajastro — APPROVED** [review]

**Pull request merged**